src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -0,0 +1,209 @@
+
+	private _fontInfo: BareFontInfo;
+	private _disassembledInstructions: WorkbenchTable<IDisassembledInstructionEntry> | null;
+
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IThemeService themeService: IThemeService,
+		@IStorageService storageService: IStorageService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService
+	) {
+		super(DISASSEMBLY_VIEW_ID, telemetryService, themeService, storageService);
+
+		this._fontInfo = BareFontInfo.createFromRawSettings(configurationService.getValue('editor'), getZoomLevel(), getPixelRatio());
+		configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration('editor')) {
+				this._fontInfo = BareFontInfo.createFromRawSettings(configurationService.getValue('editor'), getZoomLevel(), getPixelRatio());
+				this._disassembledInstructions?.rerender();
+			}
+		});
+
+		this._disassembledInstructions = null;
+	}
+
+	protected createEditor(parent: HTMLElement): void {
+		const lineHeight = this._fontInfo.lineHeight;
+		const delegate = new class implements ITableVirtualDelegate<IDisassembledInstructionEntry>{
+			headerRowHeight: number = 0; // No header
+			getHeight(row: IDisassembledInstructionEntry): number {
+				return lineHeight;
+			}
+		};
+
+		this._disassembledInstructions = this._register(this._instantiationService.createInstance(WorkbenchTable,
+			'DisassemblyView', parent, delegate,
+			[
+				{
+					label: '',
+					tooltip: '',
+					weight: 0,
+					minimumWidth: 40,
+					maximumWidth: 40,
+					templateId: BreakpointRenderer.TEMPLATE_ID,
+					project(row: IDisassembledInstructionEntry): IDisassembledInstructionEntry { return row; }
+				},
+				{
+					label: 'instructions',
+					tooltip: '',
+					weight: 0.3,
+					templateId: InstructionRenderer.TEMPLATE_ID,
+					project(row: IDisassembledInstructionEntry): IDisassembledInstructionEntry { return row; }
+				},
+			],
+			[
+				this._instantiationService.createInstance(BreakpointRenderer),
+				this._instantiationService.createInstance(InstructionRenderer),
+			],
+			{
+				identityProvider: { getId: (e: IDisassembledInstructionEntry) => e.instruction.address },
+				horizontalScrolling: false,
+				overrideStyles: {
+					listBackground: editorBackground
+				},
+				multipleSelectionSupport: false,
+				setRowLineHeight: false,
+				openOnSingleClick: false,
+			}
+		)) as WorkbenchTable<IDisassembledInstructionEntry>;
+
+		this.loadDisassembledInstructions(0, DisassemblyView.NUM_INSTRUCTIONS_TO_LOAD * 2, '0x00005000');
+		this._disassembledInstructions.reveal(DisassemblyView.NUM_INSTRUCTIONS_TO_LOAD, 0.5);
+
+		this._disassembledInstructions.onDidScroll(e => {
+			if (e.oldScrollTop > e.scrollTop && e.scrollTop < e.height) {
+				const topElement = Math.floor(e.scrollTop / this._fontInfo.lineHeight) + DisassemblyView.NUM_INSTRUCTIONS_TO_LOAD;
```

Comment:
If I comment out those two lines, nothing is rendered.
If I comment out the reveal line only, the lines are rendered, and item 0 is at the top of the view.

Screen shot during debugging:

Immediately after creating the table:
![image](https://user-images.githubusercontent.com/44103947/121262680-5f3ebb00-c869-11eb-9be4-29cfe4da4918.png)

Second time reveal is invoked (in scroll event listener):
![image](https://user-images.githubusercontent.com/44103947/121262823-944b0d80-c869-11eb-8c7d-529a6c8b8529.png)

Reply:
Ok, thanks for providing more details.
To clarify:
1) You comment out your `onDidScroll` code and nothing gets rendered because you call reveal
2) You believe this is `scrollTop` is `0`. But that is expected, the table is in the inital position and the top element is shown.

I just tried this out on my machine and it works just fine. I have simple relaxed the rule of the `Go to Disassembly` command to be always shown and the table gets nicely rendered for me (with no other changes to your source code)

Notice how the scroll bar is properly 70% through on the initial show. So I do not understand what the issue is on your side

![Screenshot 2021-06-09 at 13 21 04](https://user-images.githubusercontent.com/1926584/121345800-97a3cf00-c925-11eb-8f80-c84f83a7579b.png)

Reply:
If you still see the issue on your side I suggest that you simply make a change to the PR such that I can also reproduce the problem on my machine if possible and then I can investigate with Joao the owner of the table widget. However so far all looks good to me.

Reply:
Sorry for the confusion. ScrollTop doesn't have problem. The problem is that List.view.renderHeight is 0.

Not sure why it's not reproing on your side though. Did you start debugging VS Code from VS Code, or are you using a packaged build?

Reply:
I was running vscode out of source, not packaged. 
I will check with Joao if he has ideas why it is broken on your machine and not on mine. 
Maybe one of your colleagues could also try, just so we see if he also sees the problem on his or her machine?

Reply:
Joao had a good idea
"renderHeight = 0 usually means the client did not call `layout()`"

So make sure to call the `layout` method.
Once the table is `layouted` you should call `reveal` on it
Here's how the runtime extensions editor is doing it:

https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/extensions/browser/abstractRuntimeExtensionsEditor.ts#L458

Reply:
Well that function was copied from your code... Felix said he couldn't repro the issue either. So it's probably something wrong with my machine... Let's worry about this later if any of us see this again.

Reply:
Ok, just make sure to call `layout` please.
Also my code is 5 years old and has not been updated since. It needs a lot of improvements, it is just an example.